### ISSUE TITLE
Rearrangeable and resizable panels.

### ISSUE THREAD

**Issue by aviraldg** [vector-im/riot-web#1554]
It should be possible to rearrange and resize the left, middle and right panels that currently exist.

![image_1](https://cloud.githubusercontent.com/assets/238916/15598270/037db720-23f6-11e6-8580-32ece81a1108.png)

The behaviour wrt responsiveness (currently, the left and right panels collapse on narrower screens) needs to be decided, as well as how the current design should be updated to accommodate alternative panels available for the same location (one possible design is included above)

**Comment by aviraldg** [comment]
@dbkr Do you feel this design is appropriate? One drawback here would be the extra vertical space utilized for the icons, even if you don't have multiple panels in the same area.

**Comment by eternaleye** [comment]
@aviraldg: One thing Qt programs with similar dockables do is have a lock/unlock operation - when locked, it's like the current layout: There are no drag areas or icons, because the layout just Is What It Is. When unlocked, those appear, and the view can be rearranged. It might be easy enough to put "unlock" in settings, and "lock" up with those other buttons.

**Comment by aviraldg** [comment]
Good idea, @eternaleye. This was [discussed further on #vector-dev](https://vector.im/beta/#/room/!DdJkzRliezrwpNebLk:matrix.org/$146433833679438NtmYg:matrix.org) and @ara4n says he wants to leave design decisions to the design team, so for now I'll be using a dropdown in each panel to pick which plugin should be visible, along with resize handles (which is basically the way to do this with the least effort.) When a final design is ready, I'll redo this bit to match.

**Comment by rubo77** [comment]
Please take in mind it would be helpful also if you could arrange the video window at the top of the screen, so you can better look the partner in the eye

**Comment by rubo77** [comment]
Keywords: tiling, move, screen

**Comment by aaronraimist** [comment]
Fixed in 1.0.0

**Comment by ara4n** [comment]
tbf we haven’t fixed the rearrangeable layout yet, but hopefully this will happen

**Comment by aaronraimist** [comment]
Oops yeah I somehow skipped over rearrangeability being part of this issue
